mne/inverse_sparse/mxne_inverse.py

```diff
@@ -64,7 +64,9 @@ def _prepare_gain_column(forward, info, noise_cov, pca, depth, loose, weights,
 
     if depth is not None:
         depth_prior = np.sum(gain ** 2, axis=0) ** depth
-        source_weighting = np.sqrt(depth_prior ** -1.)
+        # Spherical leadfield can be zero at the center
+        depth_prior[depth_prior == 0.] = np.min(depth_prior[depth_prior != 0.])
```

Comment:
Checking equality to zero on floats as a result of complex computations is dangerous. Something like `np.isclose` is better, assuming you can get some handle on what "effectively zero" is numerically

Reply:
true but here it is really a true zero for a sphere model just at the center of the sphere.
we want to avoid the NaN when dividing. So only a true 0. is a problem.
I would keep it as is

Reply:
I'm not convinced...

Can you guarantee that the value numerically computes to zero when using floating point (which is probably not exact about where the center of the sphere truly is)?

Even if you do, at some point when you move some number of numerical `eps` away from this true center that yields ]zero, your computation could result in something like `1e-300`, and as soon as you divide by this you end up with something that is nearly `np.inf` (or possibly equal `np.inf` depending on the operand). So what happens in that case?

Reply:
I just saw you mentioned `NaN` instead of `inf` -- in this case do you know that the thing you're dividing into also goes to zero at the same rate? If so then it's probably okay